packages/build-utils/src/clone-env.ts

```diff
@@ -1,16 +1,29 @@
+import type { ProcessEnv } from './types';
+
 /**
  * Performs a shallow clone on the environment variable object.
  *
  * @param {Object} [currentEnv] The environment object to clone
  * @returns {Object}
  */
-export function cloneEnv(currentEnv = process.env): typeof process.env {
-  return {
+export function cloneEnv(...envs: (ProcessEnv | undefined)[]): ProcessEnv {
+  if (!envs.length) {
+    envs.push(process.env);
+  }
```

Comment:
I think we should remove this default cause and use it explicitly like `Object.assign()`. That way its clear which object is getting mutated.

Reply:
Agreed